.github/workflows/release-notes.yml

```
@@ -0,0 +1,58 @@
+name: Generate release notes
+
+on:
+  workflow_dispatch:
+    inputs:
+      bc_commit_sha:
+        description: 'Commit SHA for the BC'
+        type: string
+        required: true
+      version:
+        description: 'Version to be released (for example, `9.2.1`)'
+        type: string
+        required: true
+
+env:
+  product: Elastic Agent
+  labels: docs,forwardport-main,release,skip-changelog,Team:Docs
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  build-changelog:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v5
+        with:
+          ref: ${{ inputs.bc_commit_sha }}
+      - name: Generate release notes
+        run: |
+          GOBIN=$PWD/bin go install github.com/elastic/elastic-agent-changelog-tool@latest
+          ./bin/elastic-agent-changelog-tool build --version ${{ inputs.version }}
+          ./bin/elastic-agent-changelog-tool cleanup
+          ./bin/elastic-agent-changelog-tool render --version ${{ inputs.version }}
+      - name: Get minor
+        id: get-minor
+        uses: actions/github-script@v8
+        with:
+          script: return "${{ inputs.version }}".replace(/^(\d+\.\d+).+$/m, "$1")
+          result-encoding: string
+      - name: Open PR
+        uses: peter-evans/create-pull-request@v7
```

Use the sha commit for third-party actions:
- https://docs.github.com/en/actions/reference/security/secure-use#using-third-party-actions

> Pin actions to a full-length commit SHA
Pinning an action to a full-length commit SHA is currently the only way to use an action as an immutable release. Pinning to a particular SHA helps mitigate the risk of a bad actor adding a backdoor to the action's repository, as they would need to generate a SHA-1 collision for a valid Git object payload. When selecting a SHA, you should verify it is from the action's repository and not a repository fork.



```suggestion
        uses: peter-evans/create-pull-request@271a8d0340265f705b14b6d32b9829c1cb33d45e #v7.0.8
```